util/streamer/src/streamer_mbuf.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "streamer/streamer.h"
+
+static int
+streamer_mbuf_write(struct streamer *streamer, const void *src, size_t len)
+{
+    struct streamer_mbuf *sm;
+    int rc;
+
+    if (len > UINT16_MAX) {
+        return SYS_EINVAL;
+    }
+
+    sm = (struct streamer_mbuf *)streamer;
+    rc = os_mbuf_append(sm->om, src, len);
+    if (rc != 0) {
```

Comment:
Not sure this matters, but I was wondering what would happen if sm->om was NULL. Looking at os_mbuf_append() it wil return EINVAL in this case. Looks like we will lose that error in this case (gets converted to SYS_ENOMEM). Should this code just return the error code returned by os_mbuf_append()?

Reply:
`os_mbuf_append()` uses a different set of error codes (`OS_E[...]`), so we can't just return that.  I am not strongly opposed to checking if `sm->om` is null here, but I think it would be better to just do this check once when the streamer gets initialized (I see you have made a similar comment below!).

However, I do think we need something that converts from `OS_E` codes to `SYS_E` codes.  Other parts of the code would benefit from such a thing.  So I will add it as a separate commit in this PR.